src/algebra/char_p/exp_char.lean

```diff
@@ -0,0 +1,133 @@
+/-
+Copyright (c) 2021 Jakob Scholbach. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jakob Scholbach
+-/
+import algebra.char_p.basic
+import algebra.char_zero
+import data.nat.prime
+
+/-!
+# Exponential characteristic
+
+This file defines the exponential characteristic and establishes a few basic results relating
+it to the (ordinary characteristic).
+The definition is stated for a semiring, but the actual results are for nontrivial rings
+(as far as exponential characteristic one is concerned), respectively a ring without zero-divisors
+(for prime characteristic).
+
+## Main results
+- `exp_char`: the definition of exponential characteristic
+- `exp_char_is_prime_or_one`: the exponential characteristic is a prime or one
+- `char_eq_exp_char_iff`: the characteristic equals the exponential characteristic iff the
+  characteristic is prime
+
+## Tags
+exponential characteristic, characteristic
+-/
+
+universe u
+variables (R : Type u)
+
+section semiring
+
+variables [semiring R]
+
+/-- The definition of the exponential characteristic of a semiring. -/
+class inductive exp_char (R : Type u) [semiring R] : ℕ → Prop
+| zero [char_zero R] : exp_char 1
+| prime {q : ℕ} (hprime : q.prime) [hchar : char_p R q] : exp_char q
+
+/-- The exponential characteristic is one if the characteristic is zero. -/
+lemma exp_char_one_of_char_zero (q : ℕ) [hp : char_p R 0] [hq : exp_char R q] :
+q = 1 :=
```

Comment:
```suggestion
  q = 1 :=
```
And throughout; everything before the `:=` should be indented two spaces

Reply:
Done.